sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/numberFormatExpressions.scala

```diff
@@ -168,3 +168,159 @@ case class TryToNumber(left: Expression, right: Expression)
       newRight: Expression): TryToNumber =
     copy(left = newLeft, right = newRight)
 }
+
+/**
+ * A function that converts decimal values to strings, returning NULL if the decimal value fails to
+ * match the format string.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(numberExpr, formatExpr) - Convert `numberExpr` to a string based on the `formatExpr`.
+      Throws an exception if the conversion fails. The format can consist of the following
+      characters, case insensitive:
+        '0' or '9': Specifies an expected digit between 0 and 9. A sequence of 0 or 9 in the format
+          string matches a sequence of digits in the input value, generating a result string of the
+          same length as the corresponding sequence in the format string. The result string is
+          left-padded with zeros if the 0/9 sequence comprises more digits than the matching part of
+          the decimal value, starts with 0, and is before the decimal point. Otherwise, it is
+          padded with spaces.
+        '.' or 'D': Specifies the position of the decimal point (optional, only allowed once).
+        ',' or 'G': Specifies the position of the grouping (thousands) separator (,). There must be
+          a 0 or 9 to the left and right of each grouping separator.
+        '$': Specifies the location of the $ currency sign. This character may only be specified
+          once.
+        'S' or 'MI': Specifies the position of a '-' or '+' sign (optional, only allowed once at
+          the beginning or end of the format string). Note that 'S' prints '+' for positive values
+          but 'MI' prints a space.
+        'PR': Only allowed at the end of the format string; specifies that the result string will be
+          wrapped by angle brackets if the input value is negative.
+          ('<1>').
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(454, '999');
+       454
+      > SELECT _FUNC_(454.00, '000D00');
+       454.00
+      > SELECT _FUNC_(12454, '99G999');
+       12,454
+      > SELECT _FUNC_(78.12, '$99.99');
+       $78.12
+      > SELECT _FUNC_(-12454.8, '99G999D9S');
+       12,454.8-
+  """,
+  since = "3.3.0",
```

Comment:
I've missed the 3.3 rc cut, let's change the version to 3.4.0. The same to `try_to_char`

Reply:
Done.